chainio/txmgr/geometric/geometric.go

```diff
@@ -0,0 +1,598 @@
+	}
+	if params.GasTipMultiplierPercentage == 0 {
+		params.GasTipMultiplierPercentage = defaultParams.GasTipMultiplierPercentage
+	}
+}
+
+func NewGeometricTxnManager(
+	ethClient EthBackend,
+	wallet wallet.Wallet,
+	logger logging.Logger,
+	metrics Metrics,
+	params GeometricTxnManagerParams,
+) *GeometricTxManager {
+	fillParamsWithDefaultValues(&params)
+	return &GeometricTxManager{
+		ethClient: ethClient,
+		wallet:    wallet,
+		logger:    logger.With("component", "GeometricTxManager"),
+		metrics:   metrics,
+		params:    params,
+	}
+}
+
+// GetNoSendTxOpts This generates a noSend TransactOpts so that we can use
+// this to generate the transaction without actually sending it
+func (m *GeometricTxManager) GetNoSendTxOpts() (*bind.TransactOpts, error) {
+	ctxWithTimeout, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	from, err := m.wallet.SenderAddress(ctxWithTimeout)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get sender address: %w", err)
+	}
+	return &bind.TransactOpts{
+		From:   from,
+		NoSend: true,
+		Signer: txmgr.NoopSigner,
+	}, nil
+}
+
+func newTxnRequest(tx *types.Transaction) *txnRequest {
+	return &txnRequest{
+		tx:          tx,
+		requestedAt: time.Now(),
+		txAttempts:  make([]*transaction, 0),
+	}
+}
+
+// Send is used to sign and send a transaction to an evm chain.
+// It does gas estimation and gas price bumping to ensure the transaction gets mined,
+// but it does not do nonce management, so the tx argument must have the correct nonce already set.
+//
+// Send is blocking and safe to call concurrently, so sending multiple txs in parallel is safe.
+func (t *GeometricTxManager) Send(ctx context.Context, tx *types.Transaction) (*types.Receipt, error) {
+	return t.processTransaction(ctx, newTxnRequest(tx))
+}
+
+// ProcessTransaction sends the transaction and runs the monitoring loop which will bump the gasPrice until the tx get
+// included.
+func (t *GeometricTxManager) processTransaction(ctx context.Context, req *txnRequest) (*types.Receipt, error) {
+	t.logger.Debug("new transaction",
+		"nonce", req.tx.Nonce(), "gasFeeCap", req.tx.GasFeeCap(), "gasTipCap", req.tx.GasTipCap(),
+	)
+	t.metrics.IncrementProcessingTxCount()
+	defer t.metrics.DecrementProcessingTxCount()
+
+	var txn *types.Transaction
+	var txID wallet.TxID
+	var err error
+	retryFromFailure := 0
+	from, err := t.wallet.SenderAddress(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get sender address: %w", err)
+	}
+	for retryFromFailure < t.params.MaxSendTransactionRetry {
+		gasTipCap, err := t.estimateGasTipCap(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to estimate gas tip cap: %w", err)
+		}
+		txn, err = t.updateGasTipCap(ctx, req.tx, gasTipCap, from)
+		if err != nil {
+			return nil, fmt.Errorf("failed to update gas price: %w", err)
+		}
+		txID, err = t.wallet.SendTransaction(ctx, txn)
+		// TODO: what is this...? does it come from the fireblocks wallet?
```

Comment:
is this something we are still not sure?

Reply:
yeah this stuff I'm still not sure. @ian-shim can you help here. Would like to clean up this code or at least add comments explaining how it works, for future reference.

Reply:
I think this block of code is out of date. Check out https://github.com/Layr-Labs/eigenda/pull/473/files.
The reason why we have to explicitly convert the error is because go http client returns errors as `*url.Error` type. Note that `context.DeadlineExceeded` error cannot be matched in this case. To check if the `err` is a timeout error, we have to cover the case where the error is of type `url.Error` and check using `urlErr.Timeout()` method.

Reply:
Ahh ok I remember now. We had talked about this @shrimalmadhur, I just forgot sorry.
@ian-shim we basically just converted your explicit urlErr check to using an interface instead, which could work for more types of errors than only the url errors (any error that has a Timeout() method would be caught here).

Reply:
I still have the TODO below which I'm not sure, maybe you have an answer:
```
// TODO: why do we only retry on client or server timeouts? what about other errors?
```

Reply:
We should only retry on only transient errors but not for other errors. Timeouts are one of them. Probably there are other transient errors, but I wasn't aware of any

Reply:
Ah @ian-shim sorry I was wrong above. Just understood what you meant. Added a comment in [9752cca](https://github.com/Layr-Labs/eigensdk-go/pull/284/commits/9752ccad336f17fbcc3b1ea2bbda51caac8f82d7)
This looks fine right?

Reply:
> We should only retry on only transient errors but not for other errors. Timeouts are one of them. Probably there are other transient errors, but I wasn't aware of any
Oh sorry just saw this. I would argue that every error is *possibly* transient. Even a 500 could just be transient, and there's no way to know other than to retry, no?

Reply:
Invalid requests should not be retried for sure. 
Harder to tell if a particular 500 error is transient. What we do for EigenDA is the eth client itself retries a request with a different provider if it receives a 5xx error.